src/safeposix/syscalls/fs_calls.rs

```diff
@@ -910,6 +909,421 @@ impl Cage {
             syscall_error(Errno::ENOENT, "chdir", "the directory referred to in path does not exist")
         }
     }
+
+    //------------------------------------DUP & DUP2 SYSCALLS------------------------------------
+
+    pub fn dup_syscall(&self, fd: i32, start_desc: Option<i32>) -> i32 {
+        let mut fdtable = self.filedescriptortable.write().unwrap();
+
+        let start_fd = match start_desc {
+            Some(start_desc) => start_desc,
+            None => STARTINGFD,
+        };
+
+        //checking whether the fd exists in the file table and is higher than the starting file descriptor
+        if let Some(_) = fdtable.get(&fd) {
+            if fd >= STARTINGFD {
+                return syscall_error(Errno::EBADF, "dup", "provided file descriptor is out of range");
+            }
+
+            //error below may need to be changed -- called if error getting file descriptor
+            let nextfd = if let Some(fd) = self.get_next_fd(Some(start_fd), Some(&fdtable)) {fd} else {return syscall_error(Errno::ENFILE, "dup_syscall", "no available file descriptor number could be found");};
```

Comment:
This line is a bit too long, putting the else arm on a different line would be fine